### ISSUE TITLE
Implicit Maximum Likelihood Estimation

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#16]
# Key Idea

- Overcome **Shortcomings of GANs** through new objective formulation: ***Minimizing the Expected Distance Between a Data Point and Random Sample from Generative Model***
    - Mode Dropping/Collapse
    - Vanishing Gradient
    - Unstable Training
- Propose new framework of Generative Models: ***Precision*** and ***Recall***

# Previous Approaches

### Generative Model Topography

- Prescribed Models
    - Gaussian Mixture, Hidden Markov
- Implicit Models
    - GANs, GMMNs(Generative Moment Matching Nets)
- Hybrid Models
    - Variational Autoencoder, Autoregressive Models (e.g. WaveNet)

### Theoretical Ground of GANs

- Infinite Capacity of Discriminator
    - [O.W.] Discriminator Fails to Detect Mode Collapse
- Infinite Samples from True Data Distribution
    - [O.W.] No Guarantee that Generator will recover True Data Distribution
    - [O.W.] Gradient on Generator params all near zero at Discriminator Optimum
- Gradient Optimization converges to Global Pure-Strategy Nash Equilibrium
    - [O.W.] Local Nash Equilibrium: Perfect Discriminator and Gradient Vanishing
    - Because Supports of Data Distribution and Model Distribution is nearly Disjoint

### Reverse KL-Divergence

- Maximizing Likelihood $\equiv$ Minimizing $D_{KL} (p_{data}||p_\theta)$
- This severely penalizes assigning low density to data points (high recall)
- With Underspecified Models, Minimizing $D_{KL} (p_{\theta}||p_{data} )$ tends to work better
    - Insufficient Capacity to Cover All Modes of Data - Narrow, Concentrated on Few Modes
- In practice, we only have *Empirical Data Distribution*

    **Maximum Likelihood Estimation**

    - $D_{KL}(p_{data}||p_\theta)$ is defined and finite only when $supp(p_{data} ) \in supp(p_\theta)$
    - Model Distribution Support ***has to contain*** empirical distribution support
    - Forces ***Full Recall + relatively high density to all data points to maximize likelihood***
    - Fully make use of all data points - effective in scaling up to larger capacity models

    **Reverse KL-Divergence**

    - $supp(p_\theta) \in supp(p_{data})$, so model arbitrarily chooses few modes
    - When **Sample Quality(precision)** improves, we cannot tell if it's move along the precision-recall curve or actual shift of the curve

    # Formulation

    Let $x_1, x_2, ..., x_n$ denote actual data points, and $\hat{x}_1, \hat{x_2}, ... , \hat{x_m}$ denote generated random samples. Under some conditions,

    If $p_{\theta^*}(x_1) = .. = p_{\theta^*}(x_n)$, then

    $argmin_\theta \sum_{i=1}^n E[R_i^\theta] = argmax_\theta \sum_{i=1}^nlogp_\theta(x_i)$ 

    where $R_i ^\theta = min_{j \in [m]}||\hat{x_i}^\theta - x_i||_2^2$

    With reasonable-capacity models, density assigned to each observed data points should not differ much, so the statement is generally tight enough.

    This comes from the fact that $E[R^\theta] = \int_0^{\infin}P(R^\theta>t)dt = \int_0^\infin {[1-F^\theta(t)]} dt$

    Where $F^\theta (t)$ is the total density of $P_\theta$ in $t$ radius of a given true data point.

    (We want $F^\theta (t)$ to grow as quick as possible, assigning much density to neighborhood of given data point - intuition of MLE)

    # Experiments

    - Quantitative Evaluation : Log Likelihood
    - Qualitative Evaluation : Visual Evaluation
        - Generated Samples: MNIST, Toronto Face Dataset, CIFAR-10
        - Samples & Nearest Empirical Data : ***No Overfitting***
        - Walking in the Latent Space (Interpolation) : ***No Support Collapse, Learned Manifold***
        - Samples from different epochs : ***Stable Training***

# Remarks

- Not the SOTA output quality, but all result under ***FULL RECALL***
- Propose New Direction for Generative Models, future works in need.
